lib/segment/src/vector_storage/quantized/quantized_vectors.rs

```diff
@@ -693,13 +722,59 @@ impl QuantizedVectors {
         Ok(quantized_vectors)
     }
 
-    pub fn config_exists(path: &Path) -> bool {
-        path.join(QUANTIZED_CONFIG_PATH).exists()
+    pub fn load(
+        quantization_config: &QuantizationConfig,
+        vector_storage: &VectorStorageEnum,
+        path: &Path,
+    ) -> OperationResult<Option<Self>> {
+        let config_path = Self::get_config_path(path);
+        if config_path.exists() {
+            let config: QuantizedVectorsConfig = read_json(&config_path)?;
+            if config.storage_type == QuantizedVectorsStorageType::Immutable {
```

Comment:
Why not use this function if we already have it?

```suggestion
            if config.storage_type.is_immutable() {
```